ts/src/bitmart.ts

```diff
@@ -2222,17 +2223,22 @@ export default class bitmart extends Exchange {
             market = this.market (symbol);
             request['symbol'] = market['id'];
         }
+        let response = undefined;
         let type = undefined;
         [ type, params ] = this.handleMarketTypeAndParams ('cancelAllOrders', market, params);
-        if (type !== 'spot') {
-            throw new NotSupported (this.id + ' cancelAllOrders() does not support ' + type + ' orders, only spot orders are accepted');
-        }
-        const side = this.safeString (params, 'side');
-        if (side !== undefined) {
-            request['side'] = side;
-            params = this.omit (params, 'side');
+        if (type === 'spot') {
+            const side = this.safeString (params, 'side');
+            if (side !== undefined) {
+                request['side'] = side;
+                params = this.omit (params, 'side');
+            }
```

Comment:
@Dan-krm why doing this here? we're just transporting the `side` from `params` to `request` but that is unnecessary because in the end we merge both objects.